target_chains/solana/sdk/js/pyth_solana_receiver/README.md

```diff
@@ -0,0 +1,61 @@
+const ETH_PRICE_FEED_ID = "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace"
+
+const priceServiceConnection = new PriceServiceConnection("https://hermes.pyth.network/", { priceFeedRequestConfig: { binary: true } });
+const priceUpdateData = await priceServiceConnection.getLatestVaas([SOL_PRICE_FEED_ID, ETH_PRICE_FEED_ID]);  // Fetch off-chain price update data
+
+const pythSolanaReceiver = new PythSolanaReceiver({ connection, wallet });
+const {postInstructions, cleanupInstructions, priceFeedIdToPriceUpdateAccount} = await pythSolanaReceiver.buildPostPriceUpdateInstructions(priceUpdateData); // Get instructions to post the price update data and to cleanup the accounts later
+
+
+const myFirstPythApp = new Program<MyFirstPythApp>(IDL as MyFirstPythApp, , PublicKey.unique(), {})
+const consumerInstruction : InstructionWithEphemeralSigners = {instruction: await myFirstApp.methods.consume().accounts({ solPriceUpdate: priceFeedIdToPriceUpdateAccount[SOL_PRICE_FEED_ID], ethPriceUpdate: priceFeedIdToPriceUpdateAccount[ETH_PRICE_FEED_ID] }).instruction(), signers: []}
+consumed
+
+const transactions = pythSolanaReceiver.batchIntoVersionedTransactions([...postInstructions, consumerInstruction, ...cleanupInstructions], {}); // Put all the instructions together
+await pythSolanaReceiver.provider.sendAll(transactions);
+```
+
+Alternatively you can provide a `getInstructions` method that given the mapping from price feed id to price update account addresses, returns the intructions that you want to execute. For example:
+
+```ts
+import { Connection, PublicKey } from '@solana/web3.js';
+import { PriceServiceConnection } from '@pythnetwork/price-service-client';
+import { PythSolanaReceiver } from '@pythnetwork/pyth-solana-receiver';
+import { MyFirstPythApp, IDL } from './idl/my_first_pyth_app';
+
+
+const SOL_PRICE_FEED_ID = "0xef0d8b6fda2ceba41da15d4095d1da392a0d2f8ed0c6c7bc0f4cfac8c280b56d"
+const ETH_PRICE_FEED_ID = "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace"
+
+const priceServiceConnection = new PriceServiceConnection("https://hermes.pyth.network/", { priceFeedRequestConfig: { binary: true } });
+const priceUpdateData = await priceServiceConnection.getLatestVaas([SOL_PRICE_FEED_ID, ETH_PRICE_FEED_ID]);  // Fetch off-chain price update data
+
+
+const myFirstPythApp = new Program<MyFirstPythApp>(IDL as MyFirstPythApp, , PublicKey.unique(), {})
+const getInstructions = async (priceFeedIdToPriceUpdateAccount : Record<string, PublicKey>) => { return [{instruction: await myFirstApp.methods.consume().accounts({ solPriceUpdate: priceFeedIdToPriceUpdateAccount[SOL_PRICE_FEED_ID], ethPriceUpdate: priceFeedIdToPriceUpdateAccount[ETH_PRICE_FEED_ID] }).instruction(), signers: []}] };
+
+const pythSolanaReceiver = new PythSolanaReceiver({ connection, wallet });
+const transactions = await pythSolanaReceiver.withPriceUpdate(priceUpdateData, getInstructions, {})
+await pythSolanaReceiver.provider.sendAll(transactions);
```

Comment:
1) I think you want this to be the "more ergonomic" version right? If so, suggest putting it above the other method.

2) I think it would be cleaner if the builder worked like this:

```
let transactionBuilder = pythSolanaReceiver.newTransactionBuilder();
transactionBuilder = transactionBuilder.withPriceUpdate(priceUpdateData);
transactionBuilder = transactionBuilder.withInstruction(async (p) => {
  // inline the instructions above here
})
transactionBuilder = transactionBuilder.withCleanup();
transactionBuilder.build() // or sendAll or whatever
```

This lets people incrementally add stuff to the builder (which is sort of the whole point of a builder) instead of forcing it all into a single complicated method.

(Ofc feel free to ignore this comment or address later)

Reply:
Put the "ergonomic" way first